File: tests/fixtures/check_then_act/ts/unrelated_store_recheck_fixture.ts
// A later check on a different store does not revalidate the first one.
export async function apply(store: any, other: any, id: string) {
  const record = store.get(id);
  if (record.active) { // expect: check
    await slowCheck(id); // expect: gap await
    const cfg = other.getConfig();
    if (cfg.enabled) {
      store.update(id); // expect: write
    }
  }
}

declare function slowCheck(id: string): Promise<void>;

File: tests/fixtures/check_then_act/py/unrelated_store_recheck_fixture.py
# A later check on a different store does not revalidate the first one.
async def apply(store, other, item_id):
    record = store.get(item_id)
    if record.active:  # expect: check
        await slow_check(item_id)  # expect: gap await
        cfg = other.get_config()
        if cfg.enabled:
            store.update(item_id)  # expect: write


async def slow_check(item_id):
    pass
